lib/image/bitmap.nit

```diff
@@ -0,0 +1,259 @@
+			# start loading image data, for now assume no padding
+			for x in [0..self.height[
+			do
+				var row = new Array[Int].with_capacity(self.width)
+				for y in [0..self.width[
+				do
+					var red = fileReader.read(1)[0].ascii * 256 * 256
+					var green = fileReader.read(1)[0].ascii * 256
+					var blue = fileReader.read(1)[0].ascii
+					row.add(red + green + blue)
+				end
+				self.data.add(row)
+			end
+		end
+		fileReader.close
+	end #end of load_from_file method
+
+	# Reads in a series of bytes from the FileReader when loading a Bitmap from a file
+	# FileReader.read(1) is used due to https://github.com/privat/nit/issues/1264
+	private fun read_chars(fr: FileReader, howMany: Int): String
+	do
+		var s = ""
+		for x in [1..howMany]
+		do
+			s += fr.read(1)
+		end
+		return s
+	end
+
+	# Converts the value contained in two or four bytes into an Int. Since Nit
+	# does not have a byte type, Int is used
+	private fun get_value(array: Array[Int]): Int
+	do
+		var value = 0
+		for x in [0..array.length[ do
+			value += array[x] * 256.to_f.pow(x.to_f).to_i
+		end
+		return value
+	end
+
+	# Converts the value in an Int to four bytes. Since Nit does not have a byte
+	# type, Int is used.
+	private fun set_value(array: Array[Int], start_index: Int, value: Int)
+	do
+		array[start_index] = value.bin_and(0x000000FF)
+		array[start_index + 1] = value.rshift(8).bin_and(0x000000FF)
+		array[start_index + 2] = value.rshift(16).bin_and(0x000000FF)
+		array[start_index + 3] = value.rshift(24).bin_and(0x000000FF)
+	end
+
+	# Saves the bitmap into a file
+	fun save(path: String)
+	do
+		var fw = new FileWriter.open(path)
+		# Write bitmap header
+		for x in [0..self.bitmap_header.length[ do
+			fw.write(self.bitmap_header[x].ascii.to_s)
+		end
+		# Write dib header
+		for x in [0..self.dib_header.length[ do
+			fw.write(self.dib_header[x].ascii.to_s)
+		end
+		# Write color table (if any)
+		# Write data (no padding for now)
+		for x in [0..self.height[ do
+			var row = self.data[x]
+			for y in [0..self.width[ do
+				var pixel = row[y]
+				var red = pixel.rshift(16)
+				var green = pixel.bin_and(0x00FF00).rshift(8)
+				var blue = pixel.bin_and(0x000000FF)
+				fw.write(red.ascii.to_s)
+				fw.write(green.ascii.to_s)
+				fw.write(blue.ascii.to_s)
+			end
+		end
+		fw.close
+	end #end of save
+
+	# Converts the bitmap to grayscale by manipulating each individual pixel
+	fun grayscale
+	do
+		for x in [0..self.height[ do
+			var row = self.data[x]
+			for y in [0..self.width[ do
+				var pixel = row[y]
+				var red = pixel.rshift(16)
+				var green = pixel.bin_and(0x00FF00).rshift(8)
+				var blue = pixel.bin_and(0x000000FF)
+				var lum = (0.2126 * red.to_f + 0.7152 * green.to_f + 0.0722 * blue.to_f).to_i
+				pixel = lum * 256 * 256 + lum * 256 + lum
+				self.data[x][y] = pixel
+			end
+		end
+	end
+
+	# Sets an individual pixel, where position (0, 0) represents the top-left pixel.
+	fun set_pixel(x: Int, y: Int, color: Int)
```

Comment:
You should use `fun []=(x, y, color: Int)` here to be more nity. The usage would then be `bmp[x, y] = color`.